sips/sip-026/sip-026-clarity-defi-vault.md

```diff
@@ -0,0 +1,97 @@
+
+New vault IDs are generated incrementally starting from 0. Vaults are separated logically based on this ID with a numerical value.
+
+The SIPXXX Vault trait, `sipxxx-vault-trait`, has 3 functions. These functions
+do not update state, they are view-only and they allow for a common interface:
+## Trait functions
+### asset-contract
+
+`(asset-contract ((vault-id uint)) (response principal uint))`
+
+Returns the principal of the asset being held by the vault identified by `vault-id`.
+
+Returns the token type balance `token-id` of a specific principal `who` as an
+unsigned integer wrapped in an `ok` response. It has to respond with `u0` if the
+principal does not have a balance of the specified token or if no token with
+`token-id` exists. The function should never return an `err` response and is
+recommended to be defined as read-only.
+
+### holdings
+
+`(holdings ((vault-id uint) (asset <sip-010-trait>)) (response uint uint))`
+
+Returns the total amount of the underlying asset held by the vault identified by `vault-id`.
+
+### holdings-of
+
+`(holdings-of ((vault-id uint) (asset <sip-010-trait>) (owner principal)) (response uint uint))`
+
+Returns the total amount of the underlying asset held by the vault identified by
+their vault ID for `owner`. This is used when underlying assets are divided by vault id
+and are grouped by the principal of `owner`.
+
+If the implementation does not group the vault assets by owner, return `0`.
+
+
+```clarity
+(use-trait sip-010-trait .sip-010-trait-ft-standard.sip-010-trait)
+
+(define-trait vault-trait
+  (
+    (asset-contract (uint) (response principal uint))
+
+    (holdings (uint) (response uint uint))
+
+    (holdings-of (uint principal) (response uint uint))
+  )
+)
+```
+# Related Work
+The dicussion on a vault Solidity implementation can be found here [Forum Discussion](https://www.usenix.org/conference/atc16/technical-sessions/presentation/ali)
+
+# Backwards Compatibility
+The vault implementation was inspired by the vaults used in the Arkadiko protocol. Specifically, the Vault manager contract that is meant to abstract the data used in Arkadiko vaults. A vault trait contract can be deployed to interface with the Vault manager contract to be compliant with the standard (wrapping the original contract).
+
+# Activation
+
+This trait will be considered activated when this trait is deployed to mainnet, and 3 different implementations of the trait have been deployed to mainnet, no later than Bitcoin block 900000.
+# Reference Implementation
+The reference implementation of this SIP can be found in the following GitHub repository:
+GitHub: https://github.com/FriendsFerdinand/sips/blob/vault-standard/sips/sip-vault/sip-vault.md
```

Comment:
This is a dead link. In my opinion, because this SIP is a trait proposal, it is probably sufficient just to include the text of the trait definition here.